### PR TITLE
3.x Add a event publisher to push protected mode error log to cloud watch

### PR DESCRIPTION
**Pull request by ohchocolate**
### Description of changes
* Add a event publisher to push JSON log event to CloudWatch logs when the cluster enters protected mode

### Tests
* Manually test the code and see the log in CloudWatch
* Add a unit test named test_publish_cluster_in_protected_mode_events.

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/slurm_plugin/cluster_event_publisher.py [hunk 1]
```diff
@@ -360,6 +364,27 @@ def publish_unhealthy_node_events(self, unhealthy_nodes: List[SlurmNode]):
     #         "nodes": [{"name": "queue2-st-c5large-2"}],
     #     }
     # }
+
+    def publish_cluster_in_protected_mode_events(self):
```
Comment (chenwany) [inline src/slurm_plugin/cluster_event_publisher.py:368]: Could you add a docstring for this method?

Comment (ohchocolate) [reply src/slurm_plugin/cluster_event_publisher.py:368]: Sure!

**Comment by chenwany** [comment]
please fix the code check error

**Comment by ohchocolate** [comment]
> please fix the code check error

fixed!

**Pull request merged**